### ISSUE TITLE
Add react 17 to peer dependency list

### ISSUE THREAD

**Issue by jipis** [jakezatecky/react-dual-listbox#136]
**Describe the bug**
Installation/update of this module with npm complains because rdl lists only react 15 and 16 as peer deps. I can confirm that at least the pieces of functionality I am using of rdl are compatible with react 17.

**Reproduction steps**
Use react 17. `npm i react-dual-listbox` See the error.

**Expected behavior**
Installation of rdl works with react 17 **without** having to use the `--legacy-peer-deps` flag.

**Screenshots**
N/A

**Comment by jakezatecky** [comment]
The main reason React 17 has not been added is because this library uses Enzyme for testing, and there is no official React 17 adapter yet. There are a few paths we can proceed:

* Wait for the official React 17 adapter
* Use the unofficial React 17 adapter (becomes more attractive every day)
* Switch to React Testing Library (a significant effort; unlikely to do this)
* Add React 17 to `peerDependencies` but don't test it

Notably, installing with Yarn and React 17 will net no error (but will net a warning).

**Comment by jipis** [comment]
> * Use the unofficial React 17 adapter (becomes more attractive every day)

This is what I'm doing on my project that relies on RDL. The community hasn't given us any better option. This has almost become the de facto enzyme adapter by there not being any other option. _(Thanks, community! SMH)_

**Comment by trigunam** [comment]
> The main reason React 17 has not been added is because this library uses Enzyme for testing, and there is no official React 17 adapter yet. There are a few paths we can proceed:
> 
> * Wait for the official React 17 adapter
> * Use the unofficial React 17 adapter (becomes more attractive every day)
> * Switch to React Testing Library (a significant effort; unlikely to do this)
> * Add React 17 to `peerDependencies` but don't test it
> 
> Notably, installing with Yarn and React 17 will net no error (but will net a warning).

React 17 Enzyme adaptor is available now.
https://github.com/wojtekmaj/enzyme-adapter-react-17

Can this issue be fixed? Otherwise we need install peerDependencies with React 16 only for this package to work.

**Comment by jipis** [comment]
> > The main reason React 17 has not been added is because this library uses Enzyme for testing, and there is no official React 17 adapter yet. There are a few paths we can proceed:
...
> > * Use the unofficial React 17 adapter (becomes more attractive every day)
...

> React 17 Enzyme adapter is available now.
> https://github.com/wojtekmaj/enzyme-adapter-react-17

The link you referenced *IS* the unofficial adapter referenced above. The official adapter is still... not there yet. 🤷

**Comment by trigunam** [comment]
> > > The main reason React 17 has not been added is because this library uses Enzyme for testing, and there is no official React 17 adapter yet. There are a few paths we can proceed:
> > > ...
> > > 
> > > * Use the unofficial React 17 adapter (becomes more attractive every day)
> > >   ...
> 
> > React 17 Enzyme adapter is available now.
> > https://github.com/wojtekmaj/enzyme-adapter-react-17
> 
> The link you referenced _IS_ the unofficial adapter referenced above. The official adapter is still... not there yet. 🤷

Indeed. I tried to contribute by making changes by forking the repo, but it was not straightforward to me. Are there any instructions to make changes and run successfully so that I can try and contribute this as separate fork for people who are ok to use unofficial adapter?

**Comment by jakezatecky** [comment]
I wanted to use the official adapter, but I decided to switch to the unofficial adapter for now to reduce the annoyance for users of this package.

**Comment by trigunam** [comment]
> I wanted to use the official adapter, but I decided to switch to the unofficial adapter for now to reduce the annoyance for users of this package.

Wow, I can see that it's already something I can use. Thanks for a quick change and merge.
